Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py

```diff
@@ -850,6 +860,158 @@ def set_blob_properties_command(client: Client, args: Dict[str, Any]) -> Command
+                        + "https" + "\n"
+                        + "2021-06-08" + "\n"
+                        + sr + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "").encode('UTF-8')
+        signed_hmac_sha256 = hmac.new(base64.b64decode(account_key), StringToSign, hashlib.sha256)
+        sig = base64.b64encode(signed_hmac_sha256.digest())
+
+        token = {
+            'sp': sp,
+            'st': signedstart,
+            'se': expiry,
+            'sip': sip,
+            'spr': "https",
+            'sv': "2021-06-08",
+            'sr': sr,
+            'sig': sig
+        }
+    else:
+        StringToSign = (sp + "\n" +
+                        signedstart + "\n"
+                        + expiry + "\n"
+                        + cr + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "https" + "\n"
+                        + "2021-06-08" + "\n"
+                        + sr + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "" + "\n"
+                        + "").encode('UTF-8')
+        signed_hmac_sha256 = hmac.new(base64.b64decode(account_key), StringToSign, hashlib.sha256)
+        sig = base64.b64encode(signed_hmac_sha256.digest())
+        token = {
+            'sp': sp,
+            'st': signedstart,
+            'se': expiry,
+            'spr': "https",
+            'sv': "2021-06-08",
+            'sr': sr,
+            'sig': sig
+        }
+
+    sas_token = urllib.parse.urlencode(token)
+    return sas_token
+
+
+def check_valid_permission(string, per):
+    """
+    Check the permissions follows valid permission order.
+
+    Args:
+        string : valid permissions order
+        per : permissions given
+
+    Returns:
+        bool
+
+    """
+    l = len(per)
+    if len(string) < l:
+        return False
+    for i in range(l - 1):
+        x = per[i]
+        y = per[i + 1]
+        last = string.rindex(x)
+        first = string.index(y)
+        if last == -1 or first == -1 or last > first:
+            return False
+    return True
+
+
+def generate_sas_token(client: Client, args: dict) -> CommandResults:
+    """
+    Generate sas url for Container.
+
+    Args:
+        client (Client): Azure Blob Storage API client.
+        args (dict): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs and raw response for XSOAR.
+
+    """
+    container_name = args.get("container_name")
+    signedResource = args.get("signed_resources")
+    signedPermissions = args.get("signed_permissions")
+    valid_permissions = "racwdxltmeop"
```

Comment:
valid_permissions can be set as a constant and in this case you won't need the per variable in check_valid_permission function

Reply:
our idea is we have to validate that input permissions are in right order that will be check with valid permissions for container.